File: src/api/Falchion.Villains.Vault.Api/Enums/JobType.cs
namespace Falchion.Villains.Vault.Api.Enums;

/// <summary>
/// Represents the type of work a job performs.
/// </summary>
public enum JobType
{
	/// <summary>
	/// Scrapes and parses race results from Track Shack.
	/// </summary>
	Scrape = 0,

	/// <summary>
	/// Recalculates pass counts (kills/assassins) and regenerates race statistics.
	/// </summary>
	RecalculateStats = 1
}

File: src/app/models/enums/JobType.ts
export enum JobType {
	/** * Scrape job: Scraping race results from external sources (e.g. Track Shack) */
	Scrape = 0,
	/** * RecalculateStats job: Recalculating pass/passer breakdowns and rankings for existing results */
	RecalculateStats = 1,
}

export function getJobTypeLabel(jobType: JobType): string {
	switch (jobType) {
		case JobType.Scrape: return 'Scrape';
		case JobType.RecalculateStats: return 'Recalculate Stats';
		default: return 'Unknown';
	}
};